### DISCUSSION TITLE
Adding additional data to created records

### DISCUSSION THREAD

**Discussion by hhelmric** [codatio/codat-docs#519] [Q&A] [answered]
When we are creating a new record, say a new Customer, we add the customer's name, email, addresses, etc and then Codat will attempt to push that to the underlying accounting platform, which will then assign that Customer a unique ID. Because this is async, we don't know what that ID will be (or if it will even be successful). Presumably we could store the pushID on our own server and use a web hook to update our own records with that generated Customer id if it ever does push successfully.... but it would be simpler if we could add some additional metadata (for example, our own unique identifier for that customer). Then later we could query the Customers where `metaData.ourUniqueID == ourUniqueID` to see if that Customer record exists.

Is this possible? I see the `metaData` and `supplementalData` fields, but it seems they are reserved for other uses.

**Comment by dcoplowe** [comment, marked as answer]
Hi Harry,

Thank you for suggesting the idea of adding additional fields to the metadata of our customer data type. However, at this time, we do not have plans to implement this as it would require making this update to all data types that support creating records, as we want to ensure consistent functionality across our data types. Please note that I have added this as a suggestion to our product roadmap.

It's worth noting that the time it takes for an asynchronous create/update operation to complete can vary depending on the underlying platform. Cloud-based integrations can take any thing from a few seconds to a couple of minutes. 

If you have any further questions or need assistance with any other aspect of Codat, please feel free to reach out. I'm here to help.

Thank you for your understanding and your valuable suggestion, I've added it as a feature request.

Many thanks, 

David

**Reply by hhelmric** [reply]
Thanks David! Appreciate what you all are building
